xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/AbstractSolrCoreInitializer.java

```diff
@@ -1031,12 +996,7 @@ protected void setFieldType(FieldTypeDefinition definition, boolean add) throws
      */
     protected void commit() throws SolrException
     {
-        try {
-            this.client.commit();
-        } catch (Exception e) {
-            throw new SolrException("Failed to commit", e);
-        }
-
+        this.solrSchemaUtils.commit(getCoreName());
         // Reset the cache
         this.types = null;
         this.fields = null;
```

Comment:
This reset is not really needed anymore since you always call `solrSchemaUtils.getFields` (so nothing really check if it's null anymore).

Reply:
it avoids mistakes in case an inherited class use directly the field: it ensures that the field won't contain the previous map

Reply:
Ideally we should make this field private, but that would be an API change... wdyt?

Reply:
> but that would be an API change

Yes, I almost suggested to deleted that field actually, but then I remembered that this class was public API so that would be an API breakage, unfortunately.

Anyway, I think we need to provide a totally new helper and deprecate all that in the long run.

Reply:
> Anyway, I think we need to provide a totally new helper and deprecate all that in the long run.

Agreed.

Reply:
See: https://jira.xwiki.org/browse/XWIKI-20581